dvc/scm/git/tree.py

```diff
@@ -52,7 +53,9 @@ def dvcignore(self):
         self.use_dvcignore = True
         return ret
 
-    def open(self, path, mode="r", encoding="utf-8"):
+    def open(
+        self, path, mode="r", encoding="utf-8"
+    ):  # pylint: disable=arguments-differ
```

Comment:
These `arguments differ` will be addressed a bit later. We also need to stop setting mode and encoding like that, and instead do the same thing pathlib's Path does (we've started doing it in tests/remotes/ already).